Add filtering data in maps doc. (#3640)
opensearch-project/documentation-website: _dashboards/visualize/maps.md

## Filtering data

To show a subset of the data in the index, filter the data. You can either filter data at the layer level or draw shapes on the map to filter all layer data globally. 

### Filtering data at the layer level

To filter data at the layer level, select the layer and add a filter to it.

> The following example filters the flight destination data to display only United States destinations:

Reviewer: ```suggestion
The following example shows how to filter the flight destination data to display only United States destinations:
```

1. Select the **Save** button.
1. Select the **Update** button.

For large datasets, you may want to avoid loading data for the full map. To load data only for a specific geographic area, select **Only request data around map extent**.
{: .tip}

### Drawing shapes to filter data

> You can filter your data by drawing shapes on your map. You can also modify the filter and remove it completely. 

Reviewer 2: ```suggestion
You can filter your data by drawing shapes on your map. You can also edit, temporarily disable, exclude, negate  and remove the filter completely. 
```

> When drawing a shape on the map, draw it in a counterclockwise direction to ensure the most accurate datapoints are included. Drawing in a clockwise direction does not include the correct datapoints. According to the GeoJSON specification, polygon orientation must follow the right-hand rule regarding the area that it bounds.

Reviewer 2: Sorry for confusion, We don't have to call out that drawing clockwise direction is incorrect.

Author: @VijayanB Will keep this. Thanks!

{: .note}

You can configure the following map filter properties:

- **Filter label** Allows you to give your filter a name.
- **Spatial relation** Determines which operator to use during search. Works like a Venn diagram, but as a query. (Need to go back and check this.)

> - **Impacted layers** Whether the filter is applied the current view or another action.

Reviewer 2: Impacted layers configuration is not available . Currently all layers will be impacted, unless they disable the option to apply global filters in filter section

#### Creating a new filter using a rectangle

You can filter your data by drawing a rectangular shape on the map to show a subset of your data.

1. In the **Layers** panel, select your layer.
1. Select the **Rectangle** button on the right side of the screen to draw a rectangle. 
1. In the **Filter label** field, enter a name. 
1. Choose a spatial relation type. By default, **Intersects** is selected. See [Geographic queries]({{site.url}}{{site.baseurl}}/field-types/geo-shape/) for more information on spatial relationship types.
1. Select the **Draw Rectangle** button.
1. Draw the shape over the map area that you want to select.

> 1. If you want to disable other filters from interacting with your shape filter, select your layer from the **Layers** panel and then under the **Filters** section, deselect **Apply global filters**. 

Reviewer 2: This option is used if layers don't want to be filtered using spatial filters. Does this line conveys same?

Author: @VijayanB Will re-word to "You can use this option if you do not want the given layer to be filtered by using spatial filters."

> For large datasets, you may want to avoid loading data for the whole map. To load data only for the part of the map that is currently visible, select the **Only request data around map extent** checkbox.

Reviewer 3: ```suggestion
For large datasets, you may want to avoid loading data for the full map. To load data only for specific a geographic area, select **Only request data around map extent**.
```

Author: @vagimeli This is actually part of the previous selection for overall filtering and was accidentally selected, but not technically part of this PR. I can still include this change because it's so minor, though.

{: .tip}

### Drawing shapes to filter data

You can filter your data by drawing shapes on your map. You can also modify the filter and remove it completely. In future releases, you will be able to modify the filter shape directly on the map from the Dashboards panel.

> When drawing a shape on the map across the International Dateline, you must draw it in a counterclockwise direction. Drawing in a clockwise direction will not include the correct datapoints.

Reviewer 2: i think we don't have to call out International Dateline, we can ask use to draw counter clockwise for accuracy.

Author: @VijayanB Do we have a link to the GeoJSON specification? It may help to include it.

Reviewer 2: https://www.rfc-editor.org/rfc/rfc7946

{: .note}

You can configure the following map filter properties:

> - **Filter label** Allows you to give your filter a unique name.

Reviewer 2: ```suggestion
- **Filter label** Allows you to give your filter a name.
```

- **Spacial relation** Determines which operator to use during search. Works like a Venn diagram, but as a query. (Need to go back and check this.)

> - **Impacted layers** Whether the filter is applied to current the current view or another action.

Reviewer 3: ```suggestion
- **Impacted layers** Whether the filter is applied to the current view or another action.
```

#### Creating a new filter using a rectangle

You can filter your data by drawing a rectangular shape on the map to show a subset of your data.

1. In the **Layers** panel, select your layer.
1. Select the **Rectangle** button on the right side of the screen to draw a rectangle. 

> 1. In the **Filter label** field, give your shape a name. 

Reviewer 3: ```suggestion
1. In the **Filter label** field, enter a name. 
```

> 1. Choose a spatial relation type. By default, **Intersects** is selected. See [Geographic queries](https://opensearch.org/docs/latest/field-types/geo-shape/) for more information on spatial relationship types. [Need to fix URL format.]

Reviewer 3: ```suggestion
1. Choose a spatial relation type. By default, **Intersects** is selected. See [Geographic queries]({{site.url}}{{site.baseurl}}/field-types/geo-shape/) for more information on spatial relationship types. [Need to fix URL format.]
```

1. Select the **Draw Rectangle** button.

> 1. Draw the shape over the area on the map that you want to select.

Reviewer 3: ```suggestion
1. Draw the shape over the map area that you want to select.
```

> 1. If you want to disable other filters from interacting with your shape filter, select your layer from the **Layers** panel. Then, in the **Filters** section, deselect **Apply global filters**. 

Reviewer 3: ```suggestion
1. If you want to disable other filters from interacting with your shape filter, select your layer from the **Layers** panel and then under **Filters**, deselect **Apply global filters**. 
```

1. Select the **Update** button. This prevents [what does this prevent? Need clarification on how global filters impact this.]

[Image]

#### Creating a new filter using a polygon

> You can also filter your data by drawing a polygon shape on the map to show a subset of your data.

Reviewer 3: ```suggestion
You can filter your data by drawing a polygon shape on the map to show a subset of your data.
```

1. In the **Layers** panel, select your layer.
1. Select the **Polygon** button on the right side of the screen to draw a rectangle. 

> 1. In the **Filter label** field, give your shape a name. 

Reviewer 3: ```suggestion
1. In the **Filter label** field, enter a name. 
```

> 1. Choose a spatial relation type. By default, **Intersects** is selected.See [Geographic queries](https://opensearch.org/docs/latest/field-types/geo-shape/) for more information on spatial relationship types. [Need to fix URL format.]

Reviewer 3: ```suggestion
1. Choose a spatial relation type. By default, **Intersects** is selected.See [Geographic queries]({{site.url}}{{site.baseurl}}/field-types/geo-shape/) for more information on spatial relationship types. [Need to fix URL format.]
```

1. Select the **Draw Polygon** button.

> 1. Draw the shape over the area on the map that you want to select.

Reviewer 3: ```suggestion
1. Draw the shape over the map area that you want to select.
```

> 1. Select the point that you started with to close the polygon shape.

Reviewer 3: I don't understand this sentence. What does point mean? Are you showing this in a short demo video? Image?

Author: @vagimeli I will be illustrating this in a screenshot. "Point" refers to the location on the map where the user is clicking on the map to begin the polygon shape.

> 1. If you want to disable other filters from interacting with your shape filter, select your layer from the **Layers** panel. Then, in the **Filters** section, deselect **Apply global filters**. 

Reviewer 3: ```suggestion
1. To disable other filters from interacting with your shape filter, select your layer from the **Layers** panel, and then in the **Filters** section, deselect **Apply global filters**. 
```

1. Select the **Update** button. This prevents [what does this prevent? Need clarification on how global filters impact this.]

[Image]

#### Modifying an existing shape filter

> You can modify an existing map shape to change how your data is filtered.

Reviewer 3: ```suggestion
You can modify an existing map shape to change how data is filtered.
```

> 1. Select your filter from the top of the screen.

Reviewer 3: Are we including UI so that the user knows what we mean by "from the top if the screen"?

Author: @vagimeli That's a good point. Yes, I will be including a screenshot for this.

1. Select **Edit filter**.
1. You can also select **Exclude results** to exclude results from your filter, **Temporarily disable** to temporarily disable your filter, or **Delete** to remove your filter completely.

## Using tooltips to visualize additional data

> Document layers show geopoint and geoshape document fields as locations on the map. To add more information to the locations, you can use tooltips. For example, you may want to to show flight delay, destination weather, and destination country information in the **Flight destination** layer. Perform the following steps to configure tooltips to show additional data:

Reviewer 3: ```suggestion
Document layers show geopoint and geoshape field types as locations on the map. To add more information to the locations, you can use tooltips. For example, to show flight delay, destination weather, and destination country information in the **Flight destination** layer, perform the following steps to configure tooltips to show additional data:
```

1. Select the **Save** button.
1. Select the **Update** button.

For large datasets, you may want to avoid loading data for the full map. To load data only for a specific geographic area, select **Only request data around map extent**.
{: .tip}

### Drawing shapes to filter data

You can filter your data globally by drawing [shapes]({{site.url}}{{site.baseurl}}/field-types/geo-shape) on the map. To draw a rectangle or polygon on the map, perform the following steps:

1. Select the **Rectangle** or **Polygon** icon on the right side of the map. 
1. In the **Filter label** field, enter a name for the filter. 

> 1. Choose a spatial relation type. By default, **Intersects** is selected. See [Spatial relations]({{site.url}}{{site.baseurl}}/query-dsl/geo-and-xy/xy#spatial-relations) for more information on spatial relationship types.

Reviewer: ```suggestion
1. Choose a spatial relation type. By default, **Intersects** is selected. See [Spatial relations]({{site.url}}{{site.baseurl}}/query-dsl/geo-and-xy/xy#spatial-relations) for more information about spatial relationship types.
```

1. Draw the shape over the map area that you want to select:
  - For a rectangle, select any starting point on the map (this point becomes a rectangle vertex). Then hover (do not drag) to another point on the map and select it (this point becomes the opposite vertex).  
  - For a polygon, select any starting point on the map (this point becomes a polygon vertex) and hover (do not drag) to each subsequent vertex and select that point. Finally, make sure to select the starting point again to close the polygon, as shown in the following image.

<img src="{{site.url}}{{site.baseurl}}/images/maps/draw-shape.png" alt="Drawing a polygon on a map">

### Disabling the shape filter for a map layer

By default, the shape filter is applied globally to all layers on the map. If you want to disable your shape filter for a map layer, perform the following steps:
1. Select the layer from the **Layers** panel.
1. In the **Filters** section, deselect **Apply global filters**. 
1. Select the **Update** button. 

### Modifying an existing shape filter

To modify an existing shape filter, select your filter on the top left above the map. You can perform the following operations on an existing filter:

> - **Edit filter**: change the filter name or modify the shape's coordinates.

Reviewer 4: Done, thank you!

<img src="{{site.url}}{{site.baseurl}}/images/maps/layer-selection-panel.png" alt="Tooltip with a layer selection panel" width="450">

## Adding labels to layers

Adding a label to a layer lets you visualize additional data on the map. For example, you may want to see the origin weather in the **Flight destination** layer. Perform the following steps to add a label to the **Flight destination** layer:

1. In the **Layers** panel, select **Flight destination**.
1. In the **Style** tab, select the **Add label** checkbox.

> 1. You can choose to add a label based on a fixed text to all data points in the layer or to use a field value as label text.

Reviewer: ```suggestion
1. You can choose to add a label based on fixed text to all data points in the layer or to use a field value as the label text.
```

>   - To add a fixed text label, under **Label text**, select **Fixed** and enter label text.

Reviewer: ```suggestion
  - To add a fixed-text label, under **Label text**, select **Fixed** and enter your desired label text.
```
